### ISSUE TITLE
isSymmetric assertion in LinearOSNS.cpp when running `cylinder.py`

### ISSUE THREAD

**Issue by radarsat1** [siconos/siconos#88]
Often in the debug build, I need to disable this assertion to run a model.  I think it is related to the NEDS inertia matrix.  For instance, on the current master, `cylinder.py` in MechanicsIO examples triggers this assertion during the first collision.  On the other hand `cube.py` and `sphere.py` do not.  So I think it is likely related to non-identity inertia matrices returned by Bullet's `calculateLocalInertia`.  Is there a way we can "correct" what is returned by Bullet through some kind of normalization?  Perhaps we should at least put some kind of assertion on the matrix returned by `calculateLocalInertia` to catch the problem closer to the source.

~~~
diff --git a/kernel/src/simulationTools/LinearOSNS.cpp b/kernel/src/simulationTools/LinearOSNS.cpp
index 6e3dd2a..172aed9 100644
--- a/kernel/src/simulationTools/LinearOSNS.cpp
+++ b/kernel/src/simulationTools/LinearOSNS.cpp
@@ -329,7 +329,7 @@ void LinearOSNS::computeDiagonalInteractionBlock(const InteractionsGraph::VDescr
       //      gemm(CblasNoTrans,CblasNoTrans,1.0,*leftInteractionBlock,*work,1.0,*currentInteractionBlock);
       //*currentInteractionBlock *=h;
       DEBUG_EXPR(currentInteractionBlock->display(););
-      assert(currentInteractionBlock->isSymmetric(1e-10));
+      //assert(currentInteractionBlock->isSymmetric(1e-10));


     }
~~~

**Comment by radarsat1** [comment]
On second glance, I don't see why the output from `calculateLocalInertia` is problematic.  It outputs a 3-vector which is assigned to the diagonal elements of the inertia matrix in `NewtonEulerDS::setInertia()`, which is then copied to the lower block of the  `_I` matrix.

In `cylinder.py`, the return value of `calculateLocalInertia` is 0.335068, 0.004802, 0.335068.

I am not entirely sure why this leads to a non-symmetric `currentInteractionBlock` in `LinearOSNS::computeDiagonalInteractionBlock`.

**Comment by vacary** [comment]
Very good remarks. We are still in trouble somewhere with the inertia treatment in Siconos when it comes from bullet or more generally from a special scaling.

The assertion check that the Delassus matrix W = H^T (M_h)^-1 H , with M_h = M+ h^2\theta^2 K, is symmetric.  Theoretically, it has to be a symmetric matrix. When there is no other internal forces, the term K comes from the linearization of the gyroscopic forces. It is a skew-symmetric term that causes a very bad conditioning in the matrix  M_h. Using LAPACK for computing W results in a non symmetric matrix.

Normally, removing K for a while, with setNullifyFGyr(True) should be fixed the problem. I have to continue to investigate other symplectic scheme for the rotation.

**Comment by radarsat1** [comment]
Fixed in 5aa10140edfc03058ec0b32412d4a95c8c99d032
